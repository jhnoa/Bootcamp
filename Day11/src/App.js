// @flow
import React from 'react';
import RadioGroup from './partial/RadioGroup';
import RadioItem from './partial/RadioItem';
function App() {
  let state = {
    Group: {
      test: ['step', 'down', 'carefully!'],
      now: ['this', 'is', 'clear'],
    },
  };
  let test: Array<React$Element<*>> = [];
  for (let radioName in state.Group) {
    test.push(
      <RadioGroup>
        {React.Children.map(state.Group[radioName], (item: string) => {
          return <RadioItem name={radioName}>{item}</RadioItem>;
        })}
      </RadioGroup>,
    );
  }
  return test;

  // return (
  //   <RadioGroup name="Anon">
  //     <RadioItem name="a">ada</RadioItem>
  //     <RadioItem name="a">di</RadioItem>
  //     <RadioItem name="a">mana</RadioItem>
  //   </RadioGroup>
  // );
}

export default App;
