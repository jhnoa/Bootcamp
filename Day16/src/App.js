import React, {Component} from 'react';
import {Text, View} from 'react-native';
import Style from './Style';
import SearchBar from './Partial/SearchBar';
import NotificationBar from './Partial/NotificationBar';

type State = {
  imgUrlList: Map<string, boolean>,
};

export default class App extends Component<{}, State> {
  state = {
    urlList: new Map(),
  };
  onSearchClicked = (text) => {
    this.setState({
      urlList: this.state.urlList.set(text, false),
    });
  };
  render() {
    let test = () => {};
    return (
      <View style={Style.container}>
        <NotificationBar />
        <SearchBar onSearch={this.onSearchClicked} />
        <Text>Open up App.js to start working on your app!</Text>
        <Text>Changes you make will automatically reload.</Text>
        <Text>Shake your phone to open the developer menu.</Text>
      </View>
    );
  }
}
