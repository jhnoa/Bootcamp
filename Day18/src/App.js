import React, {Component} from 'react';
import {
  StyleSheet,
  Text,
  View,
  TouchableWithoutFeedback,
  ScrollView,
} from 'react-native';
import Expo, {Font} from 'expo';

type State = {
  fontLoaded: boolean,
  startTime: ?number,
  currentTime: number,
  lap: Array<number>,
};

export default class App extends Component<{}, State> {
  state = {
    fontLoaded: false,
    startTime: null,
    currentTime: 0,
    lap: [],
  };
  timer;
  componentDidMount() {
    this._loadFont();
  }

  render() {
    let timerView = this._timerShow();
    let buttonBar = this._buttonBarShow();
    let lapView = this._lapView();
    return (
      <View style={styles.container}>
        <View style={styles.divider}>{timerView}</View>
        <View style={styles.divider}>{[buttonBar, lapView]}</View>
      </View>
    );
  }

  _loadFont = async () => {
    await Expo.Font.loadAsync({
      spaceMono: require('./SpaceMono-Regular.ttf'),
    });
    this.setState({
      fontLoaded: true,
    });
  };
  _toMMSSSS = (time: number) => {
    let tSecond = time / 1000;
    let minutes = Math.floor(tSecond / 60);
    let seconds = (tSecond % 60).toFixed(2);

    if (minutes < 10) {
      minutes = '0' + minutes;
    }
    if (seconds < 10) {
      seconds = '0' + seconds;
    }
    return minutes + ':' + seconds;
  };
  _timerShow = () => {
    let {fontLoaded, startTime, currentTime} = this.state;
    let time = startTime != null ? currentTime - startTime : 0;
    return fontLoaded ? (
      <Text style={styles.timer}>{this._toMMSSSS(time)}</Text>
    ) : (
      <Text>{this._toMMSSSS(time)}</Text>
    );
  };
  _buttonBarShow = () => {
    let startOrStop =
      this.state.startTime != null ? (
        <View style={[styles.button, styles.red]}>
          <Text>STOP</Text>
        </View>
      ) : (
        <View style={[styles.button, styles.blue]}>
          <Text>START</Text>
        </View>
      );

    let lapPress = this.state.startTime != null ? this._lapPress : null;
    return (
      <View key={Math.random()} style={styles.buttonBar}>
        <TouchableWithoutFeedback onPress={() => lapPress()}>
          <View style={[styles.button, styles.grey]}>
            <Text>LAP</Text>
          </View>
        </TouchableWithoutFeedback>;
        <TouchableWithoutFeedback onPress={() => this._toggleTimer()}>
          {startOrStop}
        </TouchableWithoutFeedback>;
      </View>
    );
  };
  _toggleTimer = () => {
    if (this.state.startTime == null) {
      this.setState((state) => {
        timer = setInterval(() => {
          this.setState((state) => {
            return {
              ...state,
              currentTime: Date.now(),
            };
          });
        }, 33);
        return {
          ...state,
          startTime: Date.now(),
          currentTime: Date.now(),
        };
      });
    } else {
      this.setState((state) => {
        clearInterval(timer);
        return {
          ...state,
          startTime: null,
          currentTime: 0,
          lap: [],
        };
      });
    }
  };
  _lapPress = () => {
    let {lap, startTime, currentTime} = this.state;
    let laps = [...lap, currentTime - startTime];
    this.setState((state) => {
      return {
        lap: laps,
      };
    });
  };
  _lapView = () => {
    return (
      <ScrollView key={Math.random()} style={styles.lapView}>
        {this.state.lap.map((time, index) => {
          return (
            <View style={styles.lapRow} key={index}>
              <Text style={styles.lapText}>{this._toMMSSSS(time)}</Text>
            </View>
          );
        })}
      </ScrollView>
    );
  };
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#000',
    alignItems: 'stretch',
    justifyContent: 'flex-end',
  },
  timer: {
    fontFamily: 'spaceMono',
    fontSize: 60,
    textAlign: 'center',
    color: 'white',
  },
  divider: {
    borderColor: 'black',
    borderWidth: 1,
  },
  buttonBar: {
    flexDirection: 'row',
    justifyContent: 'space-around',
    padding: 20,
  },
  button: {
    width: 70,
    height: 70,
    borderRadius: 70,
    alignItems: 'center',
    justifyContent: 'center',
  },
  grey: {
    backgroundColor: 'grey',
  },
  red: {
    backgroundColor: 'red',
  },
  blue: {
    backgroundColor: 'blue',
  },
  lapView: {
    height: '30%',
  },
  lapRow: {
    alignItems: 'center',
    justifyContent: 'center',
    padding: 5,
    borderColor: 'transparent',
    borderWidth: 1,
    backgroundColor: 'grey',
  },
  lapText: {
    fontSize: 15,
  },
});
