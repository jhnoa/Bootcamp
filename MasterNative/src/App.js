import React, {Component} from 'react';
import {View, ScrollView} from 'react-native';
import Style from './Style';
import SearchBar from './Partial/SearchBar';
import NotificationBar from './Partial/NotificationBar';
import ImageView from './Partial/ImageView';
import Detail from './Partial/Detail';
import MainMenu from './Partial/MainMenu';

type ImageObject = {
  id: string,
  link: string,
  title: string,
  description: string,
};

type State = {
  searchText: string,
  favorite: Array<ImageObject>,
  pages: string,
};

export default class App extends Component<{}, State> {
  state = {
    searchText: '',
    favorite: [],
    pages: 'menu',
  };
  onSearchClicked = (text: string) => {
    this.setState({
      searchText: text,
    });
  };

  onFavorite = (text: ImageObject) => {
    let index = this.state.favorite.findIndex((s) => s === text);
    console.log(index, text);
    if (index === -1) {
      console.log('ADD');
      this.setState({
        favorite: [...this.state.favorite, text],
      });
    } else {
      console.log('REMOVE');
      let newArray = [...this.state.favorite];
      newArray.splice(index, 1);
      this.setState({
        favorite: newArray,
      });
    }
  };

  onMenuSelect = (text) => {
    this.setState({pages: text});
  };

  onBack = () => {
    this.setState({pages: 'menu'});
  };

  _showPages = (pages) => {
    if (pages === 'main') {
      return (
        <View style={Style.container}>
          <NotificationBar />
          <SearchBar onSearch={this.onSearchClicked} onBack={this.onBack} />
          <ScrollView>
            <ImageView
              searchText={this.state.searchText}
              fav={this.state.favorite}
              onFavorite={this.onFavorite}
            />
          </ScrollView>
        </View>
      );
    } else if (pages === 'detail') {
      return (
        <View style={Style.container}>
          <NotificationBar />
          <Detail images={this.state.favorite} onBack={this.onBack} />
        </View>
      );
    } else if (pages === 'menu') {
      console.log(this.state.favorite);
      return (
        <View style={Style.container}>
          <NotificationBar />
          <MainMenu onPress={this.onMenuSelect} />
        </View>
      );
    }
  };

  render() {
    console.log('APP RENDER');
    return this._showPages(this.state.pages);
  }
}
