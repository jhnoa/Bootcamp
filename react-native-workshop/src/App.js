// @flow

import React, {Component} from 'react';
import Routes from './routes';
type Props = {};

class App extends Component<Props> {
  render() {
    return <Routes />;
  }
}

export default App;
