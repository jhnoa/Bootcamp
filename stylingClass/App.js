import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
let style = {
  container: {
    display: 'flex',
    flex: 1,
    flexDirection: 'column',
    paddingTop: 20,
  },
  header: {
    height: 50,
    backgroundColor: 'white',
    borderWidth: 1,
    borderColor: 'black',
    padding: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  headBtn: {
    width: 30,
    height: 30,
    backgroundColor: 'aqua',
  },
  headText: {
    fontSize: 14,
    fontWeight: '600',
  },
  body: {
    backgroundColor: 'grey',
    borderWidth: 1,
    borderColor: 'black',
    flex: 1,
    justifyContent: 'space-between',
  },
  upperBody: {
    height: 100,
    padding: 10,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'space-between',
  },
  upperBodyContent: {
    height: 80,
    width: 80,
    backgroundColor: 'yellow',
  },
  lowerBody: {
    backgroundColor: 'maroon',
    alignItems: 'center',
    justifyContent: 'space-around',
    flex: 1,
    padding: 15,
  },
  lowerBodyTitle: {
    fontSize: 33,
    fontWeight: '700',
    color: 'skyblue',
  },
  friendListView: {
    flex: 1,
    margin: 15,
    overflow: 'scroll',
    backgroundColor: 'grey',
    borderColor: 'black',
    borderWidth: 1,
    alignSelf: 'stretch',
    padding: 20,
  },
  friend: {
    height: 80,
    padding: 15,
    flexDirection: 'row',
    alignItems: 'stretch',
    borderColor: 'black',
    borderWidth: 1,
  },
  friendPP: {
    height: 50,
    width: 50,
    borderRadius: 35,
    backgroundColor: 'white',
  },
  friendName: {
    marginHorizontal: 15,
    fontSize: 40,
  },
};

export default class App extends React.Component {
  render() {
    return (
      <View style={style.container}>
        <View style={style.header}>
          <Text style={style.headBtn}>Back</Text>
          <Text style={style.headText}>Facebook Profile</Text>
          <Text style={style.headBtn}>Share</Text>
        </View>
        <View style={style.body}>
          <View style={style.upperBody}>
            <Text style={style.upperBodyContent}>Profile Pict</Text>
            <Text style={style.upperBodyContent}>Add Friend</Text>
          </View>
          <View style={style.lowerBody}>
            <Text style={style.lowerBodyTitle}>Friend List</Text>
            <View style={style.friendListView}>
              <View style={style.friend}>
                <View style={style.friendPP} />
                <Text style={style.friendName}>Name</Text>
              </View>
              <View style={style.friend}>
                <View style={style.friendPP} />
                <Text style={style.friendName}>Name</Text>
              </View>
              <View style={style.friend}>
                <View style={style.friendPP} />
                <Text style={style.friendName}>Name</Text>
              </View>
              <View style={style.friend}>
                <View style={style.friendPP} />
                <Text style={style.friendName}>Name</Text>
              </View>
              <View style={style.friend}>
                <View style={style.friendPP} />
                <Text style={style.friendName}>Name</Text>
              </View>
              <View style={style.friend}>
                <View style={style.friendPP} />
                <Text style={style.friendName}>Name</Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    );
  }
}
