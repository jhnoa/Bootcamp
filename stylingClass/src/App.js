import React from 'react';
import {StyleSheet, Text, View} from 'react-native';
import {createStackNavigator} from 'react-navigation';
import style from './style';
import Route from './route';

export default class App extends React.Component {
  render() {
    return <Route />;
  }
}
