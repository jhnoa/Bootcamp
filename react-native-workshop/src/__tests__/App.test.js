// @flow
import React from 'react';
import App from '../App';
import renderer from 'react-test-renderer';

describe('LoginScreen', () => {
  it('should render LoginScreen corectly', () => {
    let component = renderer.create(<App />);
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
