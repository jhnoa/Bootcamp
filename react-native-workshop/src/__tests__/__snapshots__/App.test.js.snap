// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginScreen should render LoginScreen corectly 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    Test
  </Text>
</View>
`;
